### DISCUSSION TITLE
LDAP login, struggling setting default groups on users

### DISCUSSION THREAD

**Discussion by LavissaWoW** [inventree/InvenTree#5098] [Q&A]
So, I just this week set up our production server where I work to accept login via LDAP/AD.

I've set up a group that has read rights for all apps, and set its name for the SIGNUP_GROUP setting.

Creating a local user populates this group as one would expect, but User objects created via LDAP do not get the group applied to them.

Anyone have experience/advice for this?
I don't want to rely on AD groups to define the rights, as that's just too much overhead.

**Comment by matmair** [comment]
There is currently no easy way to do this, an AD Group would be the simplest solution. If you are using AD you could use the domain user group, which should be assigned to any user (if you are running 2008 R2 combat) so achives basically the same thing.

**Reply by LavissaWoW** [reply]
Piggybacking on the initial question:

Atm, I'm appending code to settings.py to run LDAP.
Are there any ways settings.py and/or Django checks for files to run as if the code was written in settings.py?

Atm, I have to insert the bit of LDAP config code into settings.py.
On update, I then have to do a git reset, pull down changes, then cat the relevant code back into settings.py.

**Reply by matmair** [reply]
That is the way to work with LDAP currently - there is no oficial support.

**Reply by matmair** [reply]
This might change if #3746 is realised.
